petab/v2/petab1to2.py

```diff
@@ -393,3 +389,73 @@ def update_noise_dist(row):
         df.drop(columns=[v1.C.OBSERVABLE_TRANSFORMATION], inplace=True)
 
     return df
+
+
+def v1v2_parameter_df(
+    parameter_df: pd.DataFrame,
+) -> pd.DataFrame:
+    """Convert parameter table from petab v1 to v2.
+
+    Do all the necessary conversions to the parameter table that can
+    be done with the parameter table alone.
+    """
+    df = parameter_df.copy().reset_index()
+
+    # parameter.estimate: int -> bool
+    df[v2.C.ESTIMATE] = df[v1.C.ESTIMATE].apply(
+        lambda x: str(bool(int(x))).lower()
+    )
+
+    def update_prior(row):
+        """Convert prior to v2 format."""
+        prior_type = row.get(v1.C.OBJECTIVE_PRIOR_TYPE)
+        if pd.isna(prior_type):
+            prior_type = v1.C.UNIFORM
+
+        pscale = row.get(v1.C.PARAMETER_SCALE)
+        if pd.isna(pscale):
+            pscale = v1.C.LIN
+
+        if (
+            pscale == v1.C.LIN
+            or prior_type not in v1.C.PARAMETER_SCALE_PRIOR_TYPES
+        ):
+            return prior_type
```

Comment:
For the weird (but valid in v1?) case of `parameterScale=lin` + `objectivePriorType=parameterScale*`, will this return an invalid v2 prior, even though a valid conversion is possible?

Reply:
You are right, thanks, will fix.